### DISCUSSION TITLE
Why do we disable SteamFix every time we install a new version of the patch?

### DISCUSSION THREAD

**Discussion by Albeoris** [Albeoris/Memoria#433] [Q&A] [answered]
@Tirlititi , do you remember why we disable SteamFix and enable it again during the patch installation?

**Comment by Tirlititi** [comment, marked as answer]
I don't know exactly how the Windows hook works, but for what I understood, you cannot replace the ``FF9_Launcher.exe`` by an updated version when that hook is enabled. At the time, I tried to do multiple things, like replacing both the normal launcher and ``FF9_Launcher.fix`` with no success.

In any case, the administrator rights for the patcher are not required / asked to people who install Memoria for the first time or who never enabled "Fix Steam Overlay" (which requires administrator rights anyways). So in my opinion, it's a non-issue and certainly not worth making 2 different versions of the patcher. People who don't trust the patcher will likely not trust the launcher either; if somehow they *previously* trusted the launcher but don't want to trust the patcher afterwards, they can just disable the fix from the launcher and then run the patcher: it won't ask for administrator rights anymore.

**Reply by Albeoris** [reply]
Hm. In fact, the main point is that if there is already a redirect to `FF9_Launcher.fix` in the registry, you don’t need to touch `FF9_Launcher.exe` at all, just replace `FF9_Launcher.fix` with the newest version.

I'm thinking in this direction because I'm a little tired of the constant messages from UAC 2 times per launch of the patcher. :D

Anyway, got it, thank you! :)

P.S. Regarding the question how is it works: when Windows run the `.exe` file it check the registry key `SOFTWARE\Microsoft\Windows NT\CurrentVersion\Image File Execution Options\{FILE_NAME.exe}`. If the key is exists and `Debugger` property is set, the value from `Debugger` will be used as a new path to `.exe` file instead of original one. So we'll run `FF9_Launcher.fix` instead of `FF9_Launcher.exe`. But any read/write operations to `FF9_Launcher.exe` will open for read/write `FF9_Launcher.exe`, so Steam integrity check will be passed successfully.

So, if you do not run `FF9_Launcher.exe` you must not get any trouble from the hook.
